devicetypes/technisat/technisat-on-off-switch.src/technisat-on-off-switch.groovy

```diff
@@ -0,0 +1,527 @@
+	if (evt.isStateChange)
+	{
+		[evt, response(["delay 3000", meterGet(scale: 2).format()])]
+	} 
+	else 
+	{
+		evt
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd)
+{
+	log.debug "Switch binary report: "+cmd
+	def value = (cmd.value ? "on" : "off")
+	createEvent(name: "switch", value: value, type: "digital", descriptionText: "$device.displayName was turned $value")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) 
+{
+	def result = []
+	def msr = String.format("%04X-%04X-%04X", cmd.manufacturerId, cmd.productTypeId, cmd.productId)
+	log.debug "msr: $msr"
+	updateDataValue("MSR", msr)
+
+	result << createEvent(descriptionText: "$device.displayName MSR: $msr", isStateChange: false)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) 
+{
+	def param = parameterMap.find( {it.paramZwaveNum == cmd.parameterNumber } )
+		
+	if(state.currentConfig."$param.key".status != "sync")
+	{
+		if(state.currentConfig."$param.key"?.newValue == cmd.scaledConfigurationValue ||
+			 state.currentConfig."$param.key".status == "init")
+		{
+			log.debug "Parameter ${param.key} set to value:${cmd.scaledConfigurationValue}"
+			state.currentConfig."$param.key".status = "sync"
+			state.currentConfig."$param.key".value = cmd.scaledConfigurationValue
+		}
+		else
+		{
+			log.debug "Parameter ${param.key} set to value failed: is:${cmd.scaledConfigurationValue} <> ${state.currentConfig."$param.key".newValue}"
+			state.currentConfig."$param.key".status = "failed"
+			syncConfig()
+		}
+	}
+	else
+	{
+		log.debug "Parameter ${param.key} update received. value:${cmd.scaledConfigurationValue}"
+		state.currentConfig."$param.key".value  = cmd.scaledConfigurationValue
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) 
+{
+	log.debug "${device.displayName}: Unhandled: $cmd"
+	[:]
+}
+
+def on() 
+{
+	encapSequence([
+		zwave.switchBinaryV1.switchBinarySet(switchValue: 0xFF),
+		zwave.switchBinaryV1.switchBinaryGet(),
+		meterGet(scale: 2)
+	], 3000)
+}
+
+def off() 
+{
+	encapSequence([
+		zwave.switchBinaryV1.switchBinarySet(switchValue: 0x00),
+		zwave.switchBinaryV1.switchBinaryGet(),
+		meterGet(scale: 2)
+	], 3000)
+}
+
+def ping() 
+{
+	log.debug "ping()"
+	refresh()
+}
+
+def poll() 
+{
+	sendHubCommand(refresh())
+}
+
+def refresh() 
+{
+	log.debug "refresh()"
+	encapSequence([
+		zwave.switchBinaryV1.switchBinaryGet(),
+		meterGet(scale: 0),
+		meterGet(scale: 2)
+	])
+}
+
+def configure() 
+{
+	log.debug "configure()"
+	def result = []
+
+	log.debug "Configure zwaveInfo: "+zwaveInfo
+
+	initStateConfigFromDevice()
+	logStateConfig()
+	result << response(encap(meterGet(scale: 0)))
+	result << response(encap(meterGet(scale: 2)))
+	result << response(encap(zwave.switchBinaryV1.switchBinaryGet()))
+	result
+}
+
+def reset() 
+{
+	encapSequence([
+		meterReset(),
+		meterGet(scale: 0)
+	])
+}
```

Comment:
Could you please remove this function? It was used by custom commands which are not supported anymore.